chembl_db/generate_chembl_db.py

```diff
@@ -0,0 +1,498 @@
+                logger.info("    progress: %d/%d (elapsed %.1fs, eta %.1fs)", processed, total, elapsed, eta)
+            else:
+                logger.info("    progress: %d/%d (elapsed %.1fs)", processed, total, elapsed)
+
+    mapping: Dict[str, Set[str]] = up.get_pfam_for_uniprot_ids(
+        proteins,
+        max_workers=max_workers,
+        timeout=timeout,
+        max_retries=retries,
+        show_progress=False,              # suppress tqdm visual
+        progress_callback=on_progress,    # push progress to logs
+    )
+    mapping_df = up.mapping_to_dataframe(mapping)
+
+    out = results.copy().merge(
+        mapping_df[["uniprot", "pfam_ids"]],
+        left_on="protein",
+        right_on="uniprot",
+        how="left",
+        validate="m:1",
+    ).drop(columns=["uniprot"])
+
+    ordered_cols = [
+    "ligand_id",
+    "pdb_id",
+    "smiles",
+    "inchikey",
+    "protein",
+    "pfam_ids",
+    "pchembl",
+    "comment",
+]
+    out = out[ordered_cols]
+    return out
+
+
+# ========================== CLI ==========================
+
+def parse_args() -> argparse.Namespace:
+    """
+    Parse command-line arguments for the pipeline.
+
+    Returns
+    -------
+    argparse.Namespace
+        Parsed arguments used by `main`.
+    """
+    ap = argparse.ArgumentParser(description="""Run ChEMBL→CCD→Pfam pipeline (logged) and export CSV. This pipeline requires a local ChEMBL SQLite database (downloadable from:
+    https://ftp.ebi.ac.uk/pub/databases/chembl/ChEMBLdb/latest/)""")
+    ap.add_argument("--chembl-sqlite", required=True, help="Path to chembl_XX.db (SQLite)")
+    ap.add_argument("--ccd-script", default="download_parse_ccd.py", help="Path to download_parse_ccd.py (default: download_parse_ccd.py)")
+    ap.add_argument("--workdir", default="temp", help="Working directory for CCD step (default: temp)")
+    ap.add_argument("--ccd-csv", default="temp/ccd.csv", help="CCD CSV output (default: temp/ccd.csv)")
+    ap.add_argument("--out-csv", default="chembl_db.csv", help="Final CSV output (default: chembl_db.csv)")
+    ap.add_argument("--skip-ccd", action="store_true", help="Skip running the CCD script (assumes --ccd-csv exists)")
+    ap.add_argument("--drop-missing-pdb", action="store_true", help="Drop rows with NaN in pdb_id after CCD merge")
+    ap.add_argument("--pfam-workers", type=int, default=24, help="Threads for UniProt Pfam step (default: 24)")
+    ap.add_argument("--pfam-timeout", type=int, default=45, help="Per-request timeout seconds (default: 45)")
+    ap.add_argument("--pfam-retries", type=int, default=3, help="Max retries for UniProt calls (default: 3)")
+    ap.add_argument("--log-dir", default="logs", help="Directory to store logs (default: logs)")
+    ap.add_argument("--log-level", default="INFO", help="Logging level (DEBUG, INFO, WARNING, ERROR)")
+    ap.add_argument("--no-console", action="store_true", help="Do not echo logs to console (file only)")
+    return ap.parse_args()
```

Comment:
No need for the argparse in this module (it is not supposed to have CLI). However, save the list of arguments in some local file for you. We will still need them in the future when Joaquin starts working on the UI